### PR TITLE
Measure View width instead of App width

### LINKED ISSUE #727: Spreadsheet view by itself overflows X causing horizontal scroll
Similar to #667

Simply go to File->Open tabular data and you'll get x scrollbar

### PR DESCRIPTION
**Pull request by garrettjstevens**
We were measuring the whole app width and passing that to the session, which had an autorun that then updated the view widths.

This simplifies things by just measuring the view widths directly and setting them, and not worrying about the whole app width at all. Since it measures the actual view widths, if a view shrinks due to a scroll bar appearing, everything updates accordingly and there isn't a problem with unwanted horizontal scroll bars.

The second commit adds a BaseViewModel for views to compose so they will be sure to have the basic requirements of a view (id, width, optional displayName, optional menuOptions).

Resolves #727

### CODE REVIEW

**Comment by cmdcolin** [comment]
Curious: Is there any way to get rid of width on our persistent state models entirely? It seems like something that should not really be part of our persisted state models

File: packages/core/ui/App.js [hunk 1]
```diff
@@ -36,28 +33,17 @@ const useStyles = makeStyles({
...
-  const { visibleDrawerWidget, appWidth, drawerWidth } = session
+  const { visibleDrawerWidget, drawerWidth } = session
 
   return (
     <div
-      ref={measureRef}
       className={classes.root}
       style={{
-        gridTemplateColumns: `[main] ${appWidth}px${
+        gridTemplateColumns: `[main] 1fr${
```
Comment (cmdcolin) [inline packages/core/ui/App.js:46]: can you elaborate on the usage of the fr unit, just something I haven't used before

Comment (garrettjstevens) [reply packages/core/ui/App.js:46]: It's the fraction of the available space. Here it just means "the rest of the space available."

https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Basic_Concepts_of_Grid_Layout#The_fr_Unit

**Comment by garrettjstevens** [comment]
> Is there any way to get rid of width on our persistent state models entirely?

We could make them be volatile, which I think makes sense since they aren't something that needs to get saved. I'll try that.

File: packages/core/ui/ViewContainer.tsx [hunk 2]
```diff
@@ -121,84 +125,96 @@ const ViewMenu = observer(
...
+      view: IBaseViewModel
+      onClose: () => void
+      style: React.CSSProperties
+      children: React.ReactNode
+      contentRect: ContentRect
+      measureRef: React.RefObject<HTMLDivElement>
+    }) => {
+      const classes = useStyles()
+      const theme = useTheme()
+      const padWidth = theme.spacing(1)
```
Comment (cmdcolin) [inline packages/core/ui/ViewContainer.tsx:147]: is padWidth required to be equal to the margin used in the theme?

**Comment by cmdcolin** [comment]
I like this PR a lot :) did a quick test and appeared to confirm no more side scrolling

**Comment by rbuels** [comment]
sounds good to me

File: packages/core/ui/ViewContainer.tsx (hunk 2 above)
Comment (garrettjstevens) [reply packages/core/ui/ViewContainer.tsx:147]: Do you mean the classes.viewContainer margin? If so, no, it could be different and this would still work.

Comment (cmdcolin) [reply packages/core/ui/ViewContainer.tsx:147]: Ok, I just wasn't wondering if the margin and padding were accidentally being called the same thing interchangeably but if they are different that is ok.

**Pull request merged**